rcl/test/rcl/test_arguments.cpp

```diff
@@ -633,29 +638,64 @@ TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), test_remove_deprecat
     &nonros_argc,
     &nonros_argv);
 
-  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
   ASSERT_EQ(nonros_argc, 5);
   EXPECT_STREQ(nonros_argv[0], "process_name");
   EXPECT_STREQ(nonros_argv[1], "-d");
   EXPECT_STREQ(nonros_argv[2], "--foo=bar");
   EXPECT_STREQ(nonros_argv[3], "--bar=baz");
   EXPECT_STREQ(nonros_argv[4], "arg");
-  EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
+}
 
-  if (NULL != nonros_argv) {
+TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), test_remove_ros_args_if_ros_only) {
+  const char * argv[] = {"--ros-args", "--disable-rosout-logs"};
+  int argc = sizeof(argv) / sizeof(const char *);
+
+  rcl_allocator_t alloc = rcl_get_default_allocator();
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+  rcl_ret_t ret = rcl_parse_arguments(argc, argv, alloc, &parsed_args);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+    EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
+  });
+
+  int nonros_argc = 0;
+  const char ** nonros_argv = NULL;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
     alloc.deallocate(nonros_argv, alloc.state);
```

Comment:
Is this deallocation correct? We're expecting `nonros_argv` to be `NULL` so I guess this is not needed.

Reply:
I think it's correct. In case `rcl_parse_arguments` isn't working correctly and allocated something, `deallocate` will work.
If the pointer is `nullptr`, deallocate is a noop, so it's still fine.

Reply:
Precisely.